Added documentation for e2e tests (#97)
kyma-project/eventing-manager: hack/e2e/README.md

# Eventing End-to-End Tests

> Tests the end-to-end flow for eventing-manager.

Reviewer: ```suggestion
Tests the end-to-end flow for Eventing manager.
```

## Overview

This test covers the end-to-end flow for Eventing. It is divided in three parts:

> 1. `setup` - prepares an Eventing CR and verifies that all the required resources are being provisioned by the eventing-manager.

Reviewer: ```suggestion
1. `setup` - prepares an Eventing CR and verifies that all the required resources are provisioned by the Eventing manager.
```

2. `eventing` - prepares Subscription CR(s) and tests the end-to-end delivery of events for different event types.
3. `cleanup` - removes the test resources and namespaces from the cluster.

> 

Reviewer: The following content does not reflect these three phases? What exactly do they refer to?

Then again, the command list at the bottom reflects the phases, but without any explanation.

Author: I have updated the section.

Reviewer: Thank you!

## Usage

### Pre-requisites

- [Go](https://go.dev/)
- [kubectl](https://kubernetes.io/docs/tasks/tools/)
- Access to Kubernetes cluster ([k3d](https://k3d.io/) / k8s)

> - Eventing-manager deployed to the cluster without any Eventing CR.

Reviewer: ```suggestion
- Eventing manager deployed to the cluster without any Eventing CR.
```

> - For EventMesh backend, the following secrets should exist in the cluster:

Reviewer: ```suggestion
- For EventMesh backend, the following secrets must exist in the cluster:
```
"should" sounds like it's optional but I guess it isn't

  - Event Mesh credentials secret by name: `eventing-backend` in namespace: `kyma-system`.
  - OAuth client secret by name: `eventing-webhook-auth` in namespace: `kyma-system`.

### How to run end-to-end tests

1. Prepare the `.env` file based on the `.env.template`.

> ```

Reviewer: In the beginning of each line of the code block, add 3 blank spaces for proper indentation below the list item.

Author: added!

KUBECONFIG=                  # Kubernetes cluster kubeconfig file path
BACKEND_TYPE=NATS            # NATS or EventMesh
MANAGER_IMAGE=               # [Optional] Container image of eventing-manager
EVENTMESH_NAMESPACE=         # [Optional] Default is: "/default/sap.kyma/tunas-develop"
```

> 2. Run the following command to set up the environment variables in your system:

Reviewer: ```suggestion
2. To set up the environment variables in your system, run:
```

> ```bash

Reviewer: In the beginning of each line of the code block, add 3 blank spaces for proper indentation below the list item.

export $(xargs < .env)
```

> 3. Run the following make target to run the whole end-to-end test suite.

Reviewer: ```suggestion
3. To run the whole end-to-end test suite, run:
```

> ```bash

Reviewer: In the beginning of each line of the code block, add 3 blank spaces for proper indentation below the list item.

Author: added!

make e2e
```

### Make targets

> 

Reviewer: Pls add an introductory sentence what to do with these commands. 
The list explains each of them individually, but gives no context why there's a whole list of them. The procedure above only mentions `make e2e` without the suffixes.

Author: Updated!

> - `make e2e-setup` - creates an Eventing CR and verifies that all the required resources are being provisioned by the eventing-manager. If an Eventing CR already exists in the cluster, then it will only update the CR if the `spec.backend.type` is different from the backend configured for tests.

Reviewer: ```suggestion
- `make e2e-setup` - creates an Eventing CR and verifies that all the required resources are provisioned by the eventing-manager. If an Eventing CR already exists in the cluster, then it will only update the CR if the `spec.backend.type` is different from the backend configured for tests.
```

> - `make e2e-eventing-setup` - prepares Subscription CR(s) and deploys a subscriber to be used in subscriptions as a sink. It will not update the Subscription CR(s) if they already exists.

Reviewer: ```suggestion
- `make e2e-eventing-setup` - prepares Subscription CR(s) and deploys a subscriber to be used in subscriptions as a sink. It does not update the Subscription CR(s) if they already exist.
```

# Eventing End-to-End Tests
Tests the end-to-end flow for Eventing manager.

## Overview

This test covers the end-to-end flow for Eventing. It tests the creation of an Eventing CR and verifies that all the required resources are provisioned by the Eventing manager. It also tests the end-to-end delivery of events for different event types.

## Usage

> ### Pre-requisites

Reviewer: ```suggestion
### Prerequisites
```

- [Go](https://go.dev/)
- [kubectl](https://kubernetes.io/docs/tasks/tools/)
- Access to Kubernetes cluster ([k3d](https://k3d.io/) / k8s)
- Eventing manager deployed to the cluster without any Eventing CR.

> - For EventMesh backend, the following secrets must exist in the cluster:

Reviewer 2: Do you have a command to set these two secrets?

Author: We need to use the script to generate the secret, as we do in release-testing.

  - Event Mesh credentials secret by name: `eventing-backend` in namespace: `kyma-system`.
  - OAuth client secret by name: `eventing-webhook-auth` in namespace: `kyma-system`.

### How to run end-to-end tests

1. Prepare the `.env` file based on the `.env.template`.

   ```
   KUBECONFIG=                  # Kubernetes cluster kubeconfig file path
   BACKEND_TYPE=NATS            # NATS or EventMesh
   MANAGER_IMAGE=               # [Optional] Container image of eventing-manager
   EVENTMESH_NAMESPACE=         # [Optional] Default is: "/default/sap.kyma/tunas-develop"
   ```

2. To set up the environment variables in your system, run:

   ```bash
   export $(xargs < .env)
   ```

3. To run the whole end-to-end test suite, run:

   ```bash
   make e2e
   ```

> ### Make targets

Reviewer: ```suggestion
### Run a single test phase
```
Switch from feature-based to goal-based perspective
